plugins/src/android/OptimoveInitProvider.java

```diff
@@ -95,4 +106,33 @@ public int update(@NonNull Uri uri, @Nullable ContentValues values, @Nullable St
         String value = resources.getString(resId);
         return TextUtils.isEmpty(value) ? null : value;
     }
+
+    private DeferredDeepLinkHandlerInterface getDDLHandler() {
+        return (Context context, DeferredDeepLinkHelper.DeepLinkResolution resolution, String link,
+                DeferredDeepLinkHelper.DeepLink data) -> {
+            try {
+                JSONObject dataJson = null;
+                if (null != data) {
+                    JSONObject deepLinkContent = new JSONObject();
+                    if (data.content != null) {
+                        deepLinkContent.put("title", data.content.title);
+                        deepLinkContent.put("description", data.content.description);
+                    }
+                    dataJson = new JSONObject();
+                    dataJson.put("data", data.data != null ? data.data : null);
+                    dataJson.put("content", deepLinkContent);
+                    dataJson.put("url", data.url);
+
+                }
+                JSONObject deepLink = new JSONObject();
+                deepLink.put("link", link);
+                deepLink.put("resolution", resolution.ordinal());
+                deepLink.put("data", dataJson);
+                OptimoveSDKPlugin.pendingDDL = deepLink;
+                OptimoveSDKPlugin.sendMessageToJs("deepLink", deepLink);
```

Comment:
a general question for Cordova. in other sdks we used to prefix message names to reduce possibility of collisions with other plugins sending java->js messages. For example, `deepLink` --> `optimove.deep-linking.deepLink`)

Any value in doing same? If yes, better to do in a separate 'finishing' PR as issue for all messages

Reply:
worth a discussion to understand the value of it

Reply:
worth a discussion to understand the value of it

Reply:
added to the final TODO list

Reply:
In general a pointer on naming of messages that model events would be to think about declaring names in a past tense, i.e. `deepLinkMatched` etc., events are 'facts', things that have happened in the state of the program rather than "commands" to instruct some other module to do something :)

Not necessary to update unless you feel strongly about it, just something to think about.